lib/src/local_backend.rs

```diff
@@ -125,7 +117,11 @@ impl Backend for LocalBackend {
 
     fn read_file(&self, _path: &RepoPath, id: &FileId) -> BackendResult<Box<dyn Read>> {
         let path = self.file_path(id);
-        let file = File::open(path).map_err(not_found_to_backend_error)?;
+        let file = File::open(path).map_err(|err| BackendError::ReadObject {
+            object_type: "file",
+            hash: id.hex(),
+            source: Box::new(err),
+        })?;
```

Comment:
Same here: I think it would be good to distinguish between a missing object and e.g. permission errors